app/src/main/java/net/sourceforge/opencamera/ui/RecSyncView.java

```diff
@@ -0,0 +1,82 @@
+package net.sourceforge.opencamera.ui;
+
+import net.sourceforge.opencamera.MainActivity;
+import net.sourceforge.opencamera.MyDebug;
+import net.sourceforge.opencamera.R;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.Button;
+import android.widget.LinearLayout;
+
+import androidx.annotation.StringRes;
+
+import com.googleresearch.capturesync.SoftwareSyncController;
+
+/**
+ * This defines the UI for the "recSync" button, that provides quick access to a
+ * range of options.
+ * <p>
+ * Requires {@link SoftwareSyncController} to be initialized (i.e. RecSync to be started)
+ */
+public class RecSyncView extends LinearLayout {
+    private static final String TAG = "RecSyncView";
+
+    public static final float ALPHA_BUTTON_SELECTED = 1.0f;
+    public static final float ALPHA_BUTTON = 0.6f; // 0.4f tends to be hard to see in bright light
+
+    private final float scale = getResources().getDisplayMetrics().density;
+    private int total_width_dp;
+
+    private final Button button_sync_settings;
+    private final Button button_align_phases;
+
+    public RecSyncView(Context context) {
+        super(context);
+
+        if (MyDebug.LOG) {
+            Log.d(TAG, "new RecSyncView: " + this);
+        }
+
+        this.setOrientation(LinearLayout.VERTICAL);
+
+        final MainActivity main_activity = (MainActivity) this.getContext();
+
+        boolean small_screen = false;
+        total_width_dp = 280;
+        int max_width_dp = main_activity.getMainUI().getMaxHeightDp(false);
+        if (total_width_dp > max_width_dp) {
+            total_width_dp = max_width_dp;
+            small_screen = true;
```

Comment:
Where it is used?

Reply:
It is used [here](https://github.com/MobileRoboticsSkoltech/OpenCamera-Sensors/blob/6c1eec2da49be9e400b4aa987550fce6ad3a6861/app/src/main/java/net/sourceforge/opencamera/ui/MainUI.java#L879) to determine the size. It is implemented the same way as it is for PopupView - a class for a similar widget as RecSyncView.

Reply:
You link give me " if( rec_sync_view != null && recSync_width > rec_sync_view.getTotalWidth()*1.2  ) {" But where is "small_screen" which I am talking about?

Reply:
Oh, I misunderstood you, thought you were talking about `total_width_dp`. Yep, `small_screen` is unused, I'll remove it.